File: 프로그래머스/unrated/181839. 주사위 게임 1/주사위 게임 1.cpp
#include <string>
#include <vector>
#include <math.h>
using namespace std;

int solution(int a, int b) {
    return a % 2 && b % 2 ? pow(a,2) + pow(b,2) : a % 2 || b % 2 ? (a + b) * 2 : abs(a - b);
}

File: 프로그래머스/0/181839. 주사위 게임 1/주사위 게임 1.java
class Solution {
    public int solution(int a, int b) {
        int answer = 0;
        
        boolean aEven = a % 2 == 0;
        boolean bEven = b % 2 == 0;
        
        if (aEven && bEven) {
            answer = Math.abs(a - b);
        } else if (!aEven && !bEven) {
            answer = (a * a) + (b * b);
        } else {
            answer = 2 * (a + b);
        }
        
        
        return answer;
    }
}